src/class-scripts.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * Scripts class
+ *
+ * @package 3.0.0
+ */
+
+declare(strict_types=1);
+
+namespace Parsely;
+
+/**
+ * Inserts the scripts and tracking code into the site's front-end
+ *
+ * @since 1.0.0
+ * @since 3.0.0 Moved from class-parsely to separate file
+ */
+class Scripts {
+	/**
+	 * Instance of Parsely class.
+	 *
+	 * @var Parsely
+	 */
+	private $parsely;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param Parsely $parsely Instance of Parsely class.
+	 */
+	public function __construct( Parsely $parsely ) {
+		$this->parsely = $parsely;
+	}
+
+	/**
+	 * Register js scripts.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return void
+	 */
+	public function run(): void {
+		add_action( 'init', array( $this, 'register_js' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_js_api' ) );
+		add_action( 'wp_enqueue_scripts', array( $this, 'load_js_tracker' ) );
+	}
+
+	/**
+	 * Register JavaScripts, if there's an API key value saved.
+	 *
+	 * @since 2.5.0
+	 *
+	 * @return void
+	 */
+	public function register_js(): void {
+		$parsely_options = $this->parsely->get_options();
+
+		if ( $this->parsely->api_key_is_missing() ) {
+			return;
+		}
+
+		wp_register_script(
+			'wp-parsely-tracker',
+			'https://cdn.parsely.com/keys/' . $parsely_options['apikey'] . '/p.js',
```

Comment:
Are we sure that the API key will always be valid (meaning it won't contain any "forbidden" characters)? I am a bit paranoid about all kinds of injection attacks :-)